### PR TITLE
CMS-Bookstore - Milestone 2

### PR DESCRIPTION
**Pull request by udberg**
In this milestone, I initialized the Redux actions, reducers, and store. Additionally, l also defined empty React components and separated presentation and container components.

I also added two container components (those which connect with the Redux store): BooksList and BooksForm
and two presentational components: App and Book.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -10,3 +10,4 @@ const MainApp = () => (
 );
 
 ReactDOM.render(<MainApp />, document.getElementById('root'));
+
```
Comment (afanimartin) [inline src/index.js]: - Wrap the `App` component in a `Provider`
```
const MainApp = () => {
<Provider store={store}>
  <App />
<Provider />
```
- Add sample book object to the initial state object
```
const initialBook = {'title': 'title', 'author': 'some author'}
super();
this.state = {
book: initialBook
}
// now add the book as a prop to App component
```

File: src/components/BookList.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            </thead>
+            <tbody>
+                {books.map((book) => (
+                    <BookContainer
+                    id={book.id}
+                    title={book.title}
+                    category={book.category}
+                    key={book.id}
+                    removeBook={removeBook}
+                    />
```
Comment (afanimartin) [inline src/components/BookList.js]: - You should be rendering the `Book` component instead. The `BookContainer` component is unnecessary and doesn't have the props, `id`, `title`, `category`, `removeBook` but the `Book`.

File: src/components/BooksForm.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class BooksForm extends Component {
```
Comment (afanimartin) [inline src/components/BooksForm.js]: - Move this component to the `src/containers` folder where it rightly makes sense
- Add a constant with book categories. Something like
```
const categories = ['code', 'comedy'], etc
```

**Review by afanimartin — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work on this PR. You did well with most requirements.
### Highlights 🌵 🌵 🌵 
- `Book` and `BookList` components well implemented
- readme looks professional

### Change(s) 🔕 🔕 🔕 
- delete `DS_Store` folder
- render the `BookList` and `BookForm` components wrapped in `div` tags in the `App` component.
```
const App = () => {
 <div>
  <BookList />
  <BookForm />
 </div>
}
```
- Move the `BookList` component to the `containers` where it rightly belongs
- Check comments for suggestions


### Martin Afani
#### Slack: whiz25

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Hi @udberg! It appears that most of the suggested changes from the last review were not implemented i.e. : 
- [x] Your `App` component renders `<h1>Hello World</h1>` instead of the expected `BookList` and `BookForm` components. ([screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/App/AppWrapBookListAndBookForm.png))
- [x] `BookList` and `BookForm` should be in your `src/containers` folder.
- [x] Add initial state and use `Provider` to wrap your `App` component in your `src/index.js` file as highlighted in the last review.

__N.B:__ Don't hesitate to reach out in case you find this feedback unclear in any way!

_Please make the necessary changes and submit another code review request._
__[Treasure Kabareebe](https://trekab.github.io)__

**Comment by rukundoeric** [comment]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯 

- ### To Improve
  - [x] [MANDATORY] When I try to start you app, it is throwing the following error...
  Screen Shot 2021-07-13 at 19 28 50


_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by afanimartin — APPROVED** [review]
## Approved 👍🏿 👍🏿 👍🏿 
Nice work making the requested change. No errors/warnings are shown in the console. Well done.


### Martin Afani
#### Slack: whiz25

**Pull request merged**